### PR TITLE
Travis: Drop custom caching and switch to node.js 8.

### LINKED ISSUE #310: Use native Travis caching system instead of our DIY cache

### PR DESCRIPTION
**Pull request by XhmikosR**
Fixes #310

### CODE REVIEW

**Comment by XhmikosR** [comment]
@cvrebert: having issues with the `path` workaround. Any ideas?

**Comment by XhmikosR** [comment]
@cvrebert: should be ready now. Just verify yourself to be safe.

**Comment by XhmikosR** [comment]
@cvrebert: it appears coveralls fails, not due to this patch, though.

**Comment by XhmikosR** [comment]
And the build isn't marked as failed, not sure if that's intentional.

**Comment by XhmikosR** [comment]
Looking into the coveralls issue, it seems it's due to https://github.com/fishbar/jscoverage/issues/35

The potential fix might be something as simple as

```js
require("coffee-script/register")
```

Should we fork jscoverage or look for alternatives?

So this leaves us the build being marked as successful.

**Comment by cvrebert** [comment]
> Should we fork jscoverage or look for alternatives?

I would warmly welcome a patch to switch to Istanbul. I'm open to proposals for other alternative tools too.
But it seems like the easiest fix would be to just add a devdependency on coffeescript and stick with jscoverage, so I'd also accept a patch for that.

File: Gemfile.lock [hunk 1]
```diff
@@ -0,0 +1,39 @@
+GEM
```
Comment (cvrebert) [inline Gemfile.lock]: Can't we just refer to the Gemfiles from our git clone of Bootstrap?

**Comment by XhmikosR** [comment]
> But it seems like the easiest fix would be to just add a devdependency on coffeescript and stick with jscoverage, so I'd also accept a patch for that.

Nope, I tried that but still fails.

File: Gemfile.lock (hunk 1 above)
Comment (XhmikosR) [inline Gemfile.lock]: That I don't know,

**Comment by cvrebert** [comment]
> not due to this patch, though.

Ah. Then open a new issue and we'll ignore the failure for now.

File: Gemfile.lock (hunk 1 above)
Comment (cvrebert) [inline Gemfile.lock]: Should just be a matter of moving the `clone` into the `install` steps and adding a couple `cp` commands after the `clone`.

Comment (XhmikosR) [inline Gemfile.lock]: It should be doable, yes. Feel free to push here if you want.

Comment (cvrebert) [inline Gemfile.lock]: Okay, added to my todo list

Comment (XhmikosR) [inline Gemfile.lock]: I might give it a go if I have time too, but I can't promise anything.

**Comment by XhmikosR** [comment]
@Johann-S: I totally forgot I had made this. I'll test it out.

**Pull request merged**